app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -155,24 +157,33 @@ class FirebaseAuthRepository : AuthRepository {
             Log.d("DeleteAccount", "User re-authenticated")
 
             // Delete the user profile and saved hikes
-            FirebaseFirestore.getInstance()
-                .collection(ProfileRepositoryFirestore.PROFILES_COLLECTION)
-                .document(user.uid)
-                .delete()
-            FirebaseFirestore.getInstance()
-                .collection(SavedHikesRepositoryFirestore.SAVED_HIKES_COLLECTION)
-                .document(user.uid)
-                .delete()
-
-            Log.d("DeleteAccount", "Profile deleted")
-
-            // Delete the user account
-            user.delete().addOnCompleteListener { task ->
-              if (task.isSuccessful) {
-                Log.d("DeleteAccount", "User account deleted")
-                onSuccess()
+            val deleteProfileTask =
+                FirebaseFirestore.getInstance()
+                    .collection(ProfileRepositoryFirestore.PROFILES_COLLECTION)
+                    .document(user.uid)
+                    .delete()
+
+            val deleteSavedHikesTask =
+                FirebaseFirestore.getInstance()
+                    .collection(SavedHikesRepositoryFirestore.SAVED_HIKES_COLLECTION)
+                    .document(user.uid)
+                    .delete()
+
+            Tasks.whenAll(deleteProfileTask, deleteSavedHikesTask).addOnCompleteListener {
+                deleteDataTask: Task<Void?> ->
```

Comment:
Warning: Even if Firebase operations are `Task<Void>`, it's still bad practice for both reasons that Sonar gives us : 
1. Wrong logic
   - Void is not equivalent to Java void but `java.lang.Void.`
2. Unnecessary platform dependency
   -  Void is a platform type available only in the Java Runtime Environment.

Second one is more than an issue for us since we are not running using the JRE but ART, which is a specifically designed runtime for Android (ART is simply short for Android RunTime). Thus, declaring `Task<Void>` or any kind of Void is an issue on non-JRE devices. The void keyword is, here, not related to Firebase tasks. If you look carefully at `.whennAll` documentation, it says that it will always return a `Task<Void>`, which is then handled by the library for the correctness of the implementation. 

Since you don't create an object, and just reuse the library created and implemented ones, there exist a simple fix. Simply remove the type annotation, which is not needed here! Then leave it in the care of the library!

Reply:
Thanks for the explanation and the quick fix!